torch/distributed/_sharded_tensor/api.py

```diff
@@ -214,6 +223,76 @@ def _init_rpc(self):
         # Barrier for all RPCs to finish on all ranks.
         rpc.api._barrier(rpc_workers)
 
+    @classmethod
+    def init_from_local_shards(
+        cls,
+        local_shards: List[torch.Tensor],
+        shards_metadata: List[ShardMetadata],
+        process_group=None
+    ):
+
+        if len(local_shards) == 0 or len(shards_metadata) == 0:
+            raise ValueError("local_shards or shards_metadata must not be empty!")
+
+        sharded_tensor = cls.__new__(cls)
+
+        # prepare initialization
+        sharded_tensor._prepare_init(process_group=process_group)
+
+        sharded_tensor._sharding_spec = EnumerableShardingSpec(shards_metadata)
+        sharded_tensor._sharding_metadata = shards_metadata
+
+        current_rank = dist.get_rank(sharded_tensor._process_group)
+
+        local_shard_idx = 0
+        num_dims = len(shards_metadata[0].shard_lengths)
+        offset_length_tuple_list_by_dim: List[List[Tuple[int, int]]] = [[] for _ in range(num_dims)]
+        # validate the metadata and add to local_shards
+        for shard_metadata in shards_metadata:  # type: ignore[attr-defined]
```

Comment:
Shouldn't we do some validation on shards_metadata? For example, what if the user specifies a remote device that does not exist?